Contributor Documentation/Environment variables.md

```diff
@@ -0,0 +1,15 @@
+# Environment variables
+
+The following environment variables can be used to control some behavior in swift-syntax.
+
+## Build time
+
+- `SWIFT_BUILD_SCRIPT_ENVIRONMENT`: Enable assertions in release builds and remove the dependency of swift-syntax on os_log.
+- `SWIFTCI_USE_LOCAL_DEPS`: Assume that all of SourceKit-LSP’s dependencies are checked out next to it and use those instead of cloning the repositories. Primarily intended for CI environments that check out related branches.
+- `SWIFTPARSER_ENABLE_ALTERNATE_TOKEN_INTROSPECTION`: Mutate the input of `assertParse` test cases. See CONTRIBUTING.md for more information.
+- `SWIFTSYNTAX_BUILD_DYNAMIC_LIBRARIES`: Build all libraries as *dynamic* instead of *automatic*. This allows us to build swift-syntax as dynamic libraries, which in turn allows us to build SourceKit-LSP using SwiftPM on Windows. Linking swift-syntax statically into sourcekit-lsp exceeds the maximum number of exported symbols on Windows.
+- `SWIFTSYNTAX_ENABLE_RAWSYNTAX_VALIDATION`: Check that the layout of the syntax tree is correct when creating or modifying nodes. See CONTRIBUTING.md for more information
```

Comment:
We could also include a link to the article from the `SwiftSyntax` target that discusses raw syntax validation for macro testing: [Sources/SwiftSyntax/Documentation.docc/RawSyntaxValidation.md](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntax/Documentation.docc/RawSyntaxValidation.md)

Reply:
That document is target for macro developers. I think the information in `CONTRIBUTING.md` match better here.